### PR TITLE
viw: init at 20171029

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/viw/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  '';
+
+  meta = with stdenv.lib; {
+    description = "VI Worsened, a fun and light clone of VI";
+    homepage = "https://github.com/lpan/viw";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ AndersonTorres ];
+  };
+}
+# TODO: optional test mem with valgrind
```
Comment (danieldk) [inline pkgs/applications/editors/viw/default.nix:33]: Stray comment

File: pkgs/applications/editors/viw/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, fetchFromGitHub, ncurses }:
+
+stdenv.mkDerivation rec{
+  pname = "viw";
+  version = "20171029";
```
Comment (danieldk) [inline pkgs/applications/editors/viw/default.nix:5]:
```suggestion
  version = "unstable-20171029";
```

File: pkgs/applications/editors/viw/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub{
```
Comment (danieldk) [inline pkgs/applications/editors/viw/default.nix:7]:
```suggestion
  src = fetchFromGitHub {
```

Formatting

**Review by danieldk — CHANGES_REQUESTED** [review]
Added some requested changes.

I also wonder if it makes sense to add this to nixpkgs. It does not seem to be maintained anymore (the last commit is ~3 years ago). Also I wonder if it has any users. Perhaps this fits the NUR better?

File: pkgs/applications/editors/viw/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  buildInputs = [ ncurses ];
+
+  buildPhase = "CC=cc make build";
+  checkPhase = ''
+    make -C tests test-command test-buffer test-state
+  '';
+  installPhase = ''
+    install -dm 755 ${placeholder "out"}/bin ${placeholder "out"}/share/doc/${pname}
+    install -m 755 viw ${placeholder "out"}/bin
+    install -m 644 README.md ${placeholder "out"}/share/doc/${pname}
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/applications/editors/viw/default.nix:23]: If it does work without placeholder, then I can change it. No problem to me.

**Comment by AndersonTorres** [comment]
It is not updated because it is very stable and simple I think. 

About the NUR: well, I need to study it! This being the case, maybe I will just reformat the expression locally and drop this PR.

**Comment by danieldk** [comment]
> It is not updated because it is very stable and simple I think.
> 
> About the NUR: well, I need to study it! This being the case, maybe I will just reformat the expression locally and drop this PR.

I didn't want to discourage submitting anything to nixpkgs. I was just not certain ;).

File: pkgs/applications/editors/viw/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    rev = "2cf317f6d82a6fa58f284074400297b6dc0f44c2";
+    sha256 = "0bnkh57v01zay6ggk0rbddaf75i48h8z06xsv33wfbjldclaljp1";
+  };
+
+  buildInputs = [ ncurses ];
+
+  buildPhase = "CC=cc make build";
+  checkPhase = ''
+    make -C tests test-command test-buffer test-state
+  '';
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/applications/editors/viw/default.nix:19]: The tests are located in the `tests` directory. How can I instruct it to checkPhase?

File: pkgs/applications/editors/viw/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  installPhase = ''
+    install -dm 755 ${placeholder "out"}/bin ${placeholder "out"}/share/doc/${pname}
+    install -m 755 viw ${placeholder "out"}/bin
+    install -m 644 README.md ${placeholder "out"}/share/doc/${pname}
+  '';
+
+  meta = with stdenv.lib; {
+    description = "VI Worsened, a fun and light clone of VI";
+    homepage = "https://github.com/lpan/viw";
+    license = licenses.gpl3;
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/applications/editors/viw/default.nix]: OK!

**Comment by AndersonTorres** [comment]
@danieldk ping!

**Review by danieldk — APPROVED** [review]
I cleaned up `installPhase` a little. LGTM.

Result of `nixpkgs-review pr 96285` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- viw
</details>

**Pull request merged**